src/pykeen/sampling/negative_sampler.py

```diff
@@ -62,7 +61,42 @@ def get_normalized_name(cls) -> str:
         """Get the normalized name of the negative sampler."""
         return normalize_string(cls.__name__, suffix=NegativeSampler.__name__)
 
+    def sample(self, positive_batch: torch.LongTensor) -> Tuple[torch.LongTensor, Optional[torch.BoolTensor]]:
+        """
+        Generate negative samples from the positive batch.
+
+        :param positive_batch: shape: (batch_size, 3)
+            The positive triples.
+
+        :return:
+            A pair (negative_batch, filter_mask) where
+
+            1. negative_batch: shape: (batch_size, num_negatives, 3)
+               The negative batch. ``negative_batch[i, :, :]`` contains the negative examples generated from
+               ``positive_batch[i, :]``.
+            2. filter_mask: shape: (batch_size, num_negatives)
+               An optional filter mask. True where negative samples are valid.
+        """
+        # create unfiltered negative batch by corruption
+        negative_batch = self.corrupt_batch(positive_batch=positive_batch)
+
+        if self.filterer is None:
+            return negative_batch, None
+
+        # If filtering is activated, all negative triples that are positive in the training dataset will be removed
+        # FIXME should this be: (negative_batch, self.filterer(negative_batch=negative_batch)) ?
```

Comment:
@mberr this is a bug now. we need to decide if we need a completely new test case for filtered, or if we should try and pack them into the current test case

Reply:
I added test for filtered and unfiltered in the base test class

Reply:
43443fb